native/acter/src/api/account/three_pid.rs

```diff
@@ -1,31 +1,88 @@
 use acter_core::events::three_pid::{ThreePidContent, ThreePidRecord};
 use anyhow::{bail, Context, Result};
-use matrix_sdk::{
-    reqwest::{ClientBuilder, StatusCode},
-    Account,
-};
-use ruma::{thirdparty::Medium, uint, ClientSecret, SessionId};
+use matrix_sdk::reqwest::{ClientBuilder, StatusCode};
+use ruma::{assign, uint};
 use ruma_client_api::{
-    account::ThirdPartyIdRemovalStatus,
+    account::{request_3pid_management_token_via_email, ThirdPartyIdRemovalStatus},
     uiaa::{AuthData, Password, UserIdentifier},
 };
+use ruma_common::{
+    thirdparty::{Medium, ThirdPartyIdentifier},
+    ClientSecret, MilliSecondsSinceUnixEpoch, OwnedClientSecret, SessionId,
+};
 use serde::Deserialize;
 use std::{collections::BTreeMap, ops::Deref};
 
-use super::{client::Client, RUNTIME};
+use super::Account;
+use crate::{api::common::clearify_error, RUNTIME};
+
+#[derive(Clone)]
+pub struct ThreePidEmailTokenResponse {
+    inner: request_3pid_management_token_via_email::v3::Response,
+    client_secret: OwnedClientSecret,
+}
+
+impl ThreePidEmailTokenResponse {
+    pub fn sid(&self) -> String {
+        self.inner.sid.to_string()
+    }
+
+    pub fn submit_url(&self) -> Option<String> {
+        self.inner.submit_url.clone()
+    }
+
+    pub fn client_secret(&self) -> String {
+        self.client_secret.to_string()
+    }
+}
 
 #[derive(Clone)]
-pub struct ThreePidManager {
-    account: Account,
-    client: Client,
+pub struct ExternalId {
+    address: String,
+    medium: Medium,
+    added_at: MilliSecondsSinceUnixEpoch,
+    validated_at: MilliSecondsSinceUnixEpoch,
+}
+
+impl ExternalId {
+    fn new(val: &ThirdPartyIdentifier) -> Self {
+        ExternalId {
+            address: val.address.clone(),
+            medium: val.medium.clone(),
+            added_at: val.added_at,
+            validated_at: val.validated_at,
+        }
+    }
```

Comment:
nitpick but as we are fully consuming anyways, we can do without the clone:
```suggestion
    fn new(val: ThirdPartyIdentifier) -> Self {
        ExternalId {
            address: val.address,
            medium: val.medium,
            added_at: val.added_at,
            validated_at: val.validated_at,
        }
    }
```

Reply:
Fixed by https://github.com/acterglobal/a3/pull/1896/commits/09694374f6d71f9613089967aa77338fff99ed91